llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -127,6 +127,26 @@ LogicalResult ScatterOp::verify() {
     return op->emitOpError("expected index depth is static");
   }
 
+  auto dimMapTy = getDimensionMap().getType().cast<ShapedType>();
+  if (dimMapTy.getRank() != 1 || dimMapTy.getNumElements() != indexDepth) {
+    return op->emitOpError("invalid number of dimension map entries ");
+  }
+
+  auto dimMap = dimensionMap();
+  for (auto it = dimMap.begin(), e = dimMap.end(); it < e; ++it) {
+    if (std::binary_search(it + 1, dimMap.end(), *it)) {
+      return op->emitOpError("dimension map contains duplicate entries");
+    }
+  }
+
+  auto originalType = getOriginalType();
+  for (auto it : llvm::enumerate(dimMap)) {
+    if (it.value() < 0 || it.value() >= originalType.getRank()) {
+      return op->emitOpError("dimension map entry invalid at entry #0")
+             << it.index();
+    }
+  }
```

Comment:
There is a static method `isInvalid` in this file, we can reuse the method.

(Maybe should rename it more descriptive.)

Reply:
Used.